pkg/controlplane/instance.go

```diff
@@ -551,7 +552,7 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 	return m, nil
 }
 
-func labelAPIServerHeartbeatFunc(identity string) lease.ProcessLeaseFunc {
+func labelAPIServerHeartbeatFunc(identity string, apiserverId string, peerBindAddress peerproxy.PeerBindAddress, peerProxy peerproxy.Interface, reconciler reconcilers.PeerEndpointReconciler) lease.ProcessLeaseFunc {
```

Comment:
I think here we just need to pass the annotation directly, and have a little helper to build it based on the specific flag vs config

Reply:
done